src/component/marker/MarkLineView.ts

```diff
@@ -305,8 +306,13 @@ class MarkLineView extends MarkerView {
         const lineDrawMap = this.markerGroupMap;
         const lineDraw = lineDrawMap.get(seriesId)
             || lineDrawMap.set(seriesId, new LineDraw());
+        const clipPath = createClipPath(
+            (seriesModel.coordinateSystem as Cartesian2D), false, seriesModel
+        );
+        this.group.setClipPath(clipPath);
```

Comment:
Should we have a rough intersection with the coord rect and the line before calling setClipPath to speed up the clipping if no lines should be skipped as suggested in my previous comment?  Each coordSys should decide if an element should be clipped (not sure if we have such methods now). For example, the polar coord should decide the intersection of a circle (the polar) and a line in the case of markLine. This is only my personal opinion, and I'm not sure if we should go so complex here in this PR.

Reply:
> I'm thinking put this after createList which can return the array of filtered markerLine, if this array is empty, the we can skip clip.

@pissang @Ovilia Just decide to use array length